client/app.jsx

```diff
@@ -0,0 +1,153 @@
+import React from 'react';
+import ReactDOM from 'react-dom';
+import Calendar from './components/Calendar.jsx';
+import rightArrow from './styles/icons/right_arrow.svg';
+import downArrow from './styles/icons/down_arrow.svg';
+import flag from './styles/icons/flag.svg';
+import question from './styles/icons/question.svg';
+import './styles/style.css';
+import $ from 'jquery';
+
+class App extends React.Component {
+  constructor() {
+    super();
+    this.today = new Date();
+    this.state = {
+      year: this.today.getFullYear(),
+      month: this.today.getMonth(),
+      requestedDates: [],
+      stage: 0,
+      price: 0,
+      cleaning: 0,
+      maxGuests: 0,
+      minStay: 0,
+      serviceFee: 0,
+      bookedDates: []
+    };
+  }
+
+  componentDidMount() {
+    $.get('/cal', (res) => {
+      let result = JSON.parse(res);
+      this.setState({
+        price: result.price,
+        cleaning: result.cleaning,
+        maxGuests: result.maxGuests,
+        minStay: result.minStay,
+        serviceFee: result.serviceFee,
+        bookedDates: result.year
+      });
+    });
+  }
+
+  handleCheckInClick(newStage) {
+    this.setState({stage: this.state.stage === newStage ? 0 : newStage});
+  }
+
+  // handles moving the calendar to the next or previous month
+  handleCalendar(i) {
+    let newMonth = this.state.month + i;
+    let newYear = this.state.year;
+    if (newMonth === 12) {
+      newMonth = 0;
+      newYear++; 
+    } else if (newMonth === -1) {
+      newMonth = 11;
+      newYear--;
+    }
+    this.setState({
+      year: newYear,
+      month: newMonth
+    });
+  }
+
+  handleMouseOver(target) {
+    // console.log(target)
+  }
+
+  handleClearDates() {
+    this.setState({
+      stage: 'in',
+      requestedDates: [],
+      year: this.today.getFullYear(),
+      month: this.today.getMonth(),
+    });
+  }
+
+  //handle date selection
+  handleDateSelect(e) {
+    let requestedDates = this.state.requestedDates;
+    let target = +e.target.innerHTML;
+    let stage = this.state.stage;
+    if (stage === 'in') {
+      stage = 'out';
+    } else if (stage === 'out') {
+      stage = 'ready';
+    }
+    if (requestedDates.length < 2) {
+      requestedDates.push(new Date(this.state.year, this.state.month, target));
+    }
+    this.setState({
+      requestedDates,
+      stage,
+    });
+  }
+
+  render() {
+    return (
+      <div id="container">
+        <div id="bookings">
+          <h3><span id="price">{`$${this.state.price}`}</span> per night</h3>
+          <hr />
+          <div className="sub-component">
```

Comment:
Would it be possible to identify this sub-component in a more descriptive way, possibly with an id tag? I would actually argue in favor of making actual functional components for the subcomponents shown here, as it would make it easier to see at a glance what the different parts of the App component represent.